rasa_sdk/endpoint.py

```diff
@@ -89,6 +89,7 @@ async def health(_) -> HTTPResponse:
     async def webhook(request: Request) -> HTTPResponse:
         """Webhook to retrieve action calls."""
         action_call = request.json
+        action_call["domain"]["headers"] = request.headers
```

Comment:
I think the ideal solution would be to pass something like `request_context` as extra parameter to `run`. However, this would require quite a couple of changes to avoid breaking things. 

In my opinion storing this information in the `tracker` would be the next best solution as from my perspective the `tracker` _tracks_ the conversation state and so it makes sense to include the rest of the tracking here. Another things: How about keep this on a branch and cutting an alpha with your changes? This would not affect the official code but you could still go forward?

Reply:
And a simple test would in my opinion be nice 🙌🏻

Reply:
I'm not following how the `request_context` parm would work. Is the idea that tracing info would only be passed if this option was enabled?

Including it in the tracker would add a lot of noise. These structures could get big and are not related to the conversation events except for tracing purposes. The tracing data would then be added to both the tracing system, tracker_store and on to the message queue.

Reply:
> I'm not following how the request_context parm would work. Is the idea that tracing info would only be passed if this option was enabled?

I meant adding an additional parameter which stores the tracing data instead of adding it to the existing `domain` parameter.

> Including it in the tracker would add a lot of noise. 

This would only be in the `tracker` within the SDK, wouldn't it? Rasa Open Source wouldn't see it as part of the tracker. In addition, only the `tracker["events"]` property is saved to tracker stores / message queue etc.